src/tools/storage-key-recipe-resolver.ts

```diff
@@ -36,16 +37,16 @@ export class StorageKeyRecipeResolver {
   /**
    * Produces resolved recipes with storage keys.
    *
-   * TODO(#4818) Add passes to assign storage keys.
    * @throws Error if recipe fails to resolve on first or second pass.
    * @yields Resolved recipes with storage keys
    */
   async resolve(): Promise<Recipe[]> {
     const recipes = [];
     for (const recipe of this.runtime.context.allRecipes) {
       this.validateHandles(recipe);
-      // TODO(#4818): use the arcId of the long running arc annotation in the recipe, if present.
-      const arc = this.runtime.newArc(this.getArcId(recipe), ramDiskStorageKeyPrefixForTest());
+      const arcId = this.findLongRunningArcId(recipe);
+      const arc = this.runtime.newArc(
+          arcId, volatileStorageKeyPrefixForTest(), arcId ? {id: Id.fromString(arcId)} : undefined);
```

Comment:
Can the first argument of `newArc` be null?

Reply:
This will happen often. Maybe we should change the loop behavior when the `arcId` is null (not long-running).

Reply:
it's fine if this is null, the name is only used in `newArc` to generate the arc id (if not passed in.. maybe we'll want to make the `name` part of the options as well, and optional. but this is outside the scope of this pr).
this will happen to ephemeral arcs (not long running). their id doesn't really matter at all, because their IDs cannot be mapped from other arcs and aren't visible.

Reply:
Cool; that makes sense to me.